### PR TITLE
Create the first version of CD pipeline with Google cloud run

### PR DESCRIPTION
**Pull request by JackHKJ**
## What is the current behavior?

TODO: This commit requires the merge of [YuYanzy:dockerization](https://github.com/YuYanzy/Education-X/tree/dockerization)
This github action authorizes, build and push the docker image to the GCP

Issue Number: EV14

Automated CI

## Pull request type

Build related changes

### CODE REVIEW

File: .github/workflows/cloudrun-docker.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+on:
+  push:
+    branches:
+      - main
+
+env:
+  PROJECT_ID: test-education-victory # TODO: update Google Cloud project id
+  GAR_LOCATION: us-central1 # TODO: update Artifact Registry location
+  REPOSITORY: quickstart-docker-repo # TODO: update Artifact Registry repository name
+  SERVICE: rmgpgab-test-education-victory-us-central1-Education-Victorytox # TODO: update Cloud Run service name
```
Comment (Windsooon) [inline .github/workflows/cloudrun-docker.yml:56]: Do you think we should put all the variables (PROJECT_ID) in Github `Environment variables` or .env?

File: .github/workflows/cloudrun-docker.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          service_account: '${{ secrets.WIF_SERVICE_ACCOUNT }}' # e.g. - my-service-account@my-project.iam.gserviceaccount.com
+
+      # NOTE: Alternative option - authentication via credentials json
+      # - name: Google Auth
+      #   id: auth
+      #   uses: 'google-github-actions/auth@v0'
+      #   with:
+      #     credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+      # BEGIN - Docker auth and build (NOTE: If you already have a container image, these Docker steps can be omitted)
```
Comment (Windsooon) [inline .github/workflows/cloudrun-docker.yml:86]: We can remove these comments.

File: .github/workflows/cloudrun-docker.yml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      # END - Docker auth and build
+
+      - name: Deploy to Cloud Run
+        id: deploy
+        uses: google-github-actions/deploy-cloudrun@v0
+        with:
+          service: ${{ env.SERVICE }}
+          region: ${{ env.REGION }}
+          # NOTE: If using a pre-built image, update the image name here
+          image: ${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REPOSITORY }}/${{ env.SERVICE }}:${{ github.sha }}
```
Comment (Windsooon) [inline .github/workflows/cloudrun-docker.yml:120]: It is more convenient to store all variables in .env or GitHub variable

**Review by Windsooon — CHANGES_REQUESTED** [review]

**Comment by Windsooon** [comment]
@JackHKJ 
1. This is CD workflow instead of CI, please change your PR title
2. Commit message can be more clear like `Create the first version of CD pipeline with Google cloud run`

File: .github/workflows/cloudrun-docker.yml (hunk 1 above)
Comment (JackHKJ) [reply .github/workflows/cloudrun-docker.yml:56]: Those are public variables that does not include any client secret

**Pull request merged**